Framework/CurveFitting/src/Algorithms/PlotPeakByLogValue.cpp

```diff
@@ -226,160 +235,161 @@ void PlotPeakByLogValue::exec() {
+      }
 
-        ifun = fit->getProperty("Function");
-        chi2 = fit->getProperty("OutputChi2overDoF");
-
-        if (createFitOutput) {
-          MatrixWorkspace_sptr outputFitWorkspace =
-              fit->getProperty("OutputWorkspace");
-          ITableWorkspace_sptr outputParamWorkspace =
-              fit->getProperty("OutputParameters");
-          ITableWorkspace_sptr outputCovarianceWorkspace =
-              fit->getProperty("OutputNormalisedCovarianceMatrix");
-          fitWorkspaces.emplace_back(outputFitWorkspace);
-          parameterWorkspaces.emplace_back(outputParamWorkspace);
-          covarianceWorkspaces.emplace_back(outputCovarianceWorkspace);
-        }
-        g_log.debug() << "Fit result " << fit->getPropertyValue("OutputStatus")
-                      << ' ' << chi2 << '\n';
-      } catch (...) {
-        g_log.error("Error in Fit ChildAlgorithm");
-        throw;
+      g_log.debug() << "Fitting " << data.ws->getName() << " index " << data.i
+                    << " with \n";
+      g_log.debug() << ifun->asString() << '\n';
+
+      const std::string spectrum_index = std::to_string(data.i);
+      std::string wsBaseName;
+
+      if (createFitOutput)
+        wsBaseName = wsNames[i].name + "_" + spectrum_index;
+
+      bool histogramFit = getPropertyValue("EvaluationType") == "Histogram";
+      bool ignoreInvalidData = getProperty("IgnoreInvalidData");
+
+      // Fit the function
+      auto fit = this->createChildAlgorithm("Fit");
+      fit->initialize();
+      fit->setPropertyValue("EvaluationType",
+                            getPropertyValue("EvaluationType"));
+      fit->setProperty("Function", ifun);
+      fit->setProperty("InputWorkspace", data.ws);
+      fit->setProperty("WorkspaceIndex", data.i);
+      fit->setPropertyValue("StartX", getPropertyValue("StartX"));
+      fit->setPropertyValue("EndX", getPropertyValue("EndX"));
+      fit->setProperty("IgnoreInvalidData", ignoreInvalidData);
+      fit->setPropertyValue(
+          "Minimizer", getMinimizerString(wsNames[i].name, spectrum_index));
+      fit->setPropertyValue("CostFunction", getPropertyValue("CostFunction"));
+      fit->setPropertyValue("MaxIterations", getPropertyValue("MaxIterations"));
+      fit->setPropertyValue("PeakRadius", getPropertyValue("PeakRadius"));
+      fit->setProperty("CalcErrors", true);
+      fit->setProperty("CreateOutput", createFitOutput);
+      if (!histogramFit) {
+        fit->setProperty("OutputCompositeMembers", outputCompositeMembers);
+        fit->setProperty("ConvolveMembers", outputConvolvedMembers);
+        fit->setProperty("Exclude", exclude);
       }
+      fit->setProperty("Output", wsBaseName);
+      fit->execute();
 
-      // Extract the fitted parameters and put them into the result table
-      TableRow row = result->appendRow();
-      if (isDataName) {
-        row << wsNames[i].name;
-      } else {
-        row << logValue;
+      if (!fit->isExecuted()) {
```

Comment:
Does the child algorithm throw exceptions if it fails, if so use:
`fit->setRethrows(true)`
and let the exception propagate upwards without checking isExecuted

Reply:
I think it does yes. I have removed this check and the containing try catch block.